### PR TITLE
Upgrade gem for Rails 7 compatibility

### PR DESCRIPTION
**Pull request by geetfun**
This PR upgrades the gem to work with Rails 7. 

Notes:
- The existing code base works without any modifications
- Some regex were modified in the specs to accommodate for `Tempfile` directories other than /tmp. Not sure if it is due to Ruby 2.7 (requirement for Rails 7)
- Bumped semver

### CODE REVIEW

**Comment by Laykou** [comment]
@ahukkanen Could this be merged please and released under new version?

File: spec/lib/ratonvirus/storage/active_storage_spec.rb [hunk 1]
```diff
@@ -213,7 +213,7 @@
 
         it "yields with with the result of `path` of the attachable" do
           expect { |b| subject.asset_path(asset, &b) }.to yield_with_args(
-            %r{/tmp/Ratonvirus[0-9]+-[0-9]+-[0-9a-z]+\.pdf}
+            %r{/*?\/Ratonvirus[0-9]+-[0-9]+-[0-9a-z]+\.pdf}
```
Comment (ahukkanen) [inline spec/lib/ratonvirus/storage/active_storage_spec.rb:216]: Can you revert this back to `/tmp`?

File: spec/lib/ratonvirus/storage/active_storage_spec.rb [hunk 2]
```diff
@@ -223,7 +223,7 @@
...
```
Comment (ahukkanen) [inline spec/lib/ratonvirus/storage/active_storage_spec.rb:226]: Can you revert this back to `/tmp`?

File: spec/lib/ratonvirus/storage/active_storage_spec.rb [hunk 3]
```diff
@@ -246,7 +246,7 @@
           expect(attachable).to receive(:open)
 
           expect { |b| subject.asset_path(asset, &b) }.to yield_with_args(
-            %r{/tmp/RatonTest[0-9]+-[0-9]+-[0-9a-z]+\.txt}
+            %r{/*?\/RatonTest[0-9]+-[0-9]+-[0-9a-z]+\.txt}
```
Comment (ahukkanen) [inline spec/lib/ratonvirus/storage/active_storage_spec.rb:249]: Can you revert this back to `/tmp`?

File: spec/lib/ratonvirus/storage/carrierwave_spec.rb [hunk 4]
```diff
@@ -80,7 +80,7 @@
       it "yields with with a temp path of the copy of the asset" do
         expect(asset).to receive(:file).exactly(4).times.and_return(clean_file)
         expect { |b| subject.asset_path(asset, &b) }.to yield_with_args(
-          %r{/tmp/Ratonvirus[0-9]+-[0-9]+-[0-9a-z]+\.pdf}
+          %r{/*?\/Ratonvirus[0-9]+-[0-9]+-[0-9a-z]+\.pdf}
```
Comment (ahukkanen) [inline spec/lib/ratonvirus/storage/carrierwave_spec.rb:83]: Can you revert this back to `/tmp`?

File: spec/lib/ratonvirus/storage/carrierwave_spec.rb [hunk 5]
```diff
@@ -106,7 +106,7 @@
         it "yields with with a temp path of the copy of the asset" do
           expect(asset).to receive(:file).exactly(4).times.and_return(fog_file)
           expect { |b| subject.asset_path(asset, &b) }.to yield_with_args(
-            %r{/tmp/Ratonvirus[0-9]+-[0-9]+-[0-9a-z]+\.pdf}
+            %r{/*?\/Ratonvirus[0-9]+-[0-9]+-[0-9a-z]+\.pdf}
```
Comment (ahukkanen) [inline spec/lib/ratonvirus/storage/carrierwave_spec.rb:109]: Can you revert this back to `/tmp`?

**Review by ahukkanen — CHANGES_REQUESTED** [review]
I don't see the necessity for the `tmp` related changes which are also causing rubocop violations.

Can you also run rubocop and fix the violations that it provides.

Please also change the `config.load_defaults` to match the targeted Rails version at the dummy app.

**Comment by ahukkanen** [comment]
It seems also that Rubocop is giving a `parser` related warning due to the mismatch of Ruby version.

Please also run `bundle update parser` and commit the updated `Gemfile.lock`. This fixes the warning.

**Comment by geetfun** [comment]
> It seems also that Rubocop is giving a `parser` related warning due to the mismatch of Ruby version.
> 
> Please also run `bundle update parser` and commit the updated `Gemfile.lock`. This fixes the warning.

Sounds good. I'll make those changes and update the PR. Thanks

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,15 @@
...
+
+Support for Rails 7
+
+Changed:
+
+- The minimum Ruby version is now set to 2.7 as it is a requirement of Rails 7
+
+Fixed:
+
+- Some test regular expressions
```
Comment (ahukkanen) [inline CHANGELOG.md:11]: Can you take this CHANGELOG note off since it's not accurate anymore?

**Review by ahukkanen — CHANGES_REQUESTED** [review]
Thank you @geetfun !

Few more things
> Please also change the config.load_defaults to match the targeted Rails version at the dummy app.
https://github.com/mainio/ratonvirus/blob/b4eb864f14b250d0926ff84476e2ed48c39f7ef9/spec/rails_dummy_app/config/application.rb#L11

Also left one comment regarding the CHANGELOG.

**Comment by geetfun** [comment]
@ahukkanen Changes updated. Thanks!

**Review by ahukkanen — APPROVED** [review]
Thanks a lot, LGTM!

**Pull request merged**